juju/framework.py

```diff
@@ -734,3 +774,19 @@ def _from_iterable(cls, it):
         new instances from an iterable argument.
         """
         return set(it)
+
+    def __le__(self, other):
+        if isinstance(other, StoredSet):
+            return self._under <= other._under
+        elif isinstance(other, collections.abc.Set):
+            return self._under <= other
+        else:
+            return NotImplemented
+
+    def __ge__(self, other):
+        if isinstance(other, StoredSet):
+            return self._under >= other._under
+        elif isinstance(other, collections.abc.Set):
+            return self._under >= other
+        else:
+            return NotImplemented
```

Comment:
Would make sense to have `__eq__` as well, with the same rationale of the other methods (performance).

Same for Dict above. Please see if there are others as I haven't done an exhaustive check.